tests/test_routes.py

```diff
@@ -147,14 +148,63 @@ def test_not_found(self):
         self.assertEqual(self.client.get("/not-found-url").status_code, 404)
 
     def test_engage_index(self):
-        """
-        When given a /engage URL,
-        we should return a 200 status code
-        This also tests that discourse module
-        for engage pages work
-        """
+        response = self.client.get("/engage")
+        self.assertEqual(response.status_code, 200)
+
+    def test_engage_index_without_preview_flag_cannot_see_inactive_page(self):
+        response = self.client.get("/engage")
+        self.assertEqual(response.status_code, 200)
+
+        soup = BeautifulSoup(response.data, "html.parser")
+        self.assertIsNone(
+            soup.find("a", {"href": "/engage/it/deployment-azienda-manuale"})
+        )
+        self.assertIsNone(
+            soup.find("meta", {"name": "robots", "content": "noindex"})
+        )
+
+    def test_engage_index_with_preview_flag_sees_inactive_pages(self):
+        response = self.client.get("/engage?preview")
+        self.assertEqual(response.status_code, 200)
 
-        self.assertEqual(self.client.get("/engage").status_code, 200)
+        soup = BeautifulSoup(response.data, "html.parser")
+        self.assertIsNotNone(
+            soup.find(
+                "a", {"href": "/engage/it/deployment-azienda-manuale?preview"}
+            )
+        )
+        self.assertIsNotNone(
+            soup.find("meta", {"name": "robots", "content": "noindex"})
+        )
+
+    def test_active_page_returns_200(self):
+        response = self.client.get("/engage/finance")
```

Comment:
This one is about to change so will fail

Reply:
No idea what happened with the comment I dropped yesterday... 

The test only needs the page to load one time, after that it will always hit a cached copy of that page, so the test will still work. But I did change the url to check for `engage/micro-clouds` to keep it closer to 'reality'.